tests_e2e/tests/ext_sequencing/ext_sequencing.py

```diff
@@ -0,0 +1,275 @@
+#
+
+#
+# This test adds extensions with multiple dependencies to a VMSS using the 'provisionAfterExtensions' property and
+# validates they are enabled in order of dependencies.
+#
+import copy
+import uuid
+from datetime import datetime
+from typing import List, Dict, Any
+
+from assertpy import fail
+from azure.mgmt.compute.models import VirtualMachineScaleSetVMExtensionsSummary
+
+from tests_e2e.tests.ext_sequencing.ext_seq_test_cases import add_one_dependent_ext_without_settings, add_two_extensions_with_dependencies, \
+    remove_one_dependent_extension, remove_all_dependencies, add_one_dependent_extension, \
+    add_single_dependencies, remove_all_dependent_extensions, add_failing_dependent_extension_with_one_dependency, add_failing_dependent_extension_with_two_dependencies
+from tests_e2e.tests.lib.agent_test import AgentVmssTest, TestSkipped
+from tests_e2e.tests.lib.vm_extension_identifier import VmExtensionIds
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.resource_group_client import ResourceGroupClient
+from tests_e2e.tests.lib.ssh_client import SshClient
+
+
+class ExtSequencing(AgentVmssTest):
+    # Cases to test different dependency scenarios
+    test_cases = [
+        add_one_dependent_ext_without_settings,
+        add_two_extensions_with_dependencies,
+        remove_one_dependent_extension,
+        remove_all_dependencies,
+        add_one_dependent_extension,
+        add_single_dependencies,
+        remove_all_dependent_extensions,
+        add_failing_dependent_extension_with_one_dependency,
+        add_failing_dependent_extension_with_two_dependencies
+    ]
+
+    @staticmethod
+    def get_dependency_map(extensions: List[Dict[str, Any]]) -> Dict[str, Dict[str, Any]]:
+        dependency_map: Dict[str, Dict[str, Any]] = dict()
+
+        for ext in extensions:
+            ext_name = ext['name']
+            provisioned_after = ext['properties'].get('provisionAfterExtensions')
+            depends_on = provisioned_after if provisioned_after else []
+            # We know an extension should fail if a script was provided
+            ext_settings = ext['properties'].get("settings")
+            should_fail = True if ext_settings and "script" in ext_settings else False
+            dependency_map[ext_name] = {"should_fail": should_fail, "depends_on": depends_on}
+
+        return dependency_map
+
+    @staticmethod
+    def get_sorted_extension_names(extensions: List[VirtualMachineScaleSetVMExtensionsSummary], ssh_client: SshClient) -> List[str]:
+        # Using VmExtensionIds to get publisher for each ext to be used in remote script
+        extension_full_names = {
+            "AzureMonitorLinuxAgent": VmExtensionIds.AzureMonitorLinuxAgent,
+            "RunCommandLinux": VmExtensionIds.RunCommand,
+            "CustomScript": VmExtensionIds.CustomScript
+        }
+        enabled_times = []
+        for ext in extensions:
+            # Only check extensions which succeeded provisioning
```

Comment:
Because we only except a log that the extension was enabled if it succeeded provisioning. For all other extensions that did not succeed, we check that they were expected to fail on lines 124-132

Reply:
got it, thanks